node/src/accountant/db_access_objects/failed_payable_dao.rs

```diff
@@ -0,0 +1,778 @@
+        .collect()
+    }
+
+    fn insert_new_records(&self, txs: &[FailedTx]) -> Result<(), FailedPayableDaoError> {
+        if txs.is_empty() {
+            return Err(FailedPayableDaoError::EmptyInput);
+        }
+
+        let unique_hashes: HashSet<TxHash> = txs.iter().map(|tx| tx.hash).collect();
+        if unique_hashes.len() != txs.len() {
+            return Err(FailedPayableDaoError::InvalidInput(
+                "Duplicate hashes found in the input".to_string(),
+            ));
+        }
+
+        if !self.get_tx_identifiers(&unique_hashes).is_empty() {
+            return Err(FailedPayableDaoError::InvalidInput(
+                "Input hash is already present in the database".to_string(),
+            ));
+        }
+
+        let sql = format!(
+            "INSERT INTO failed_payable (\
+             tx_hash, \
+             receiver_address, \
+             amount_high_b, \
+             amount_low_b, \
+             timestamp, \
+             gas_price_wei_high_b, \
+             gas_price_wei_low_b, \
+             nonce, \
+             reason, \
+             checked
+             ) VALUES {}",
+            comma_joined_stringifiable(txs, |tx| {
+                let amount_checked = checked_conversion::<u128, i128>(tx.amount);
+                let gas_price_wei_checked = checked_conversion::<u128, i128>(tx.gas_price_wei);
+                let (amount_high_b, amount_low_b) = BigIntDivider::deconstruct(amount_checked);
+                let (gas_price_wei_high_b, gas_price_wei_low_b) =
+                    BigIntDivider::deconstruct(gas_price_wei_checked);
+                format!(
+                    "('{:?}', '{:?}', {}, {}, {}, {}, {}, {}, '{:?}', {})",
+                    tx.hash,
+                    tx.receiver_address,
+                    amount_high_b,
+                    amount_low_b,
+                    tx.timestamp,
+                    gas_price_wei_high_b,
+                    gas_price_wei_low_b,
+                    tx.nonce,
+                    tx.reason,
+                    tx.checked
+                )
+            })
+        );
+
+        match self.conn.prepare(&sql).expect("Internal error").execute([]) {
+            Ok(inserted_rows) => {
+                if inserted_rows == txs.len() {
+                    Ok(())
+                } else {
+                    Err(FailedPayableDaoError::PartialExecution(format!(
+                        "Only {} out of {} records inserted",
+                        inserted_rows,
+                        txs.len()
+                    )))
+                }
+            }
+            Err(e) => Err(FailedPayableDaoError::SqlExecutionFailed(e.to_string())),
+        }
+    }
+
+    fn retrieve_txs(&self, condition: Option<FailureRetrieveCondition>) -> Vec<FailedTx> {
+        let raw_sql = "SELECT tx_hash, receiver_address, amount_high_b, amount_low_b, \
+        timestamp, gas_price_wei_high_b, gas_price_wei_low_b, nonce, reason, checked FROM failed_payable"
```

Comment:
This SQL is a bit hard to read. Could you try to rearrange it a bit, maybe even with indentation, keywords on a different level than the values?